src/main/java/org/privacyidea/PrivacyIDEA.java

```diff
@@ -380,6 +380,9 @@ private void retrieveAuthToken()
 
         // Count down the latch to indicate that the token is retrieved
         authTokenLatch.countDown();
+
+        // Create a new CountDownLatch for the next token retrieval
+        authTokenLatch = new CountDownLatch(1);
```

Comment:
i think this should be at the start of the function

Reply:
By the first time there is already one created at the start. After use it prepares authTokenLatch for another use. Is it wrong?

Reply:
[81b52af](https://github.com/privacyidea/java-client/pull/70/commits/81b52af0309185c5e8710e1e24f5871a44cd34e9)